backend_plugin/src/test/java/BlockCommentEvaluatorTest.java

```diff
@@ -71,46 +76,208 @@ public void consecutiveLinesDownCommentedOut() {
+			event = createDocEvent(offset, SINGLE_SLASH);
+			assertFalse(testEvaluator.evaluate(event));
+
+			// Place another backslash after the previous one
+			offset++;
+			event = createDocEvent(offset, SINGLE_SLASH);
+			// Now the evaluation function should trigger
+			assertTrue(testEvaluator.evaluate(event));
+
+			// Comment out the third line
+			// Place a single backslash at the start of the third line
+			offset = doc.getLineOffset(2);
+			event = createDocEvent(offset, SINGLE_SLASH);
+			assertFalse(testEvaluator.evaluate(event));
+
+			// Place another backslash after the previous one
+			offset++;
+			event = createDocEvent(offset, SINGLE_SLASH);
+			// Now the evaluation function should trigger
+			assertTrue(testEvaluator.evaluate(event));
+		} catch (BadLocationException e) {
+			// Should never get here
+			fail("Should never see this error in: " + this.getClass().getSimpleName() + "::" + this.getClass().getName());
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Tests the block comment evaluation function returns true when line 3 then line 2 then line 1 are commented out
+	 */
+	@Test
+	public void threeConsecutiveLinesUpCommentedOut() {
+		try {
+			// Mock a document event with a single backslash placed at the beginning of the third line
+			offset = doc.getLineOffset(2);
+			event = createDocEvent(offset, SINGLE_SLASH);
+			assertFalse(testEvaluator.evaluate(event));
+
+			// Place a second backslash after the first
+			offset++;
+			event = createDocEvent(offset, SINGLE_SLASH);
+			assertFalse(testEvaluator.evaluate(event));
+
+			// Mock a document event with a single backslash placed at the beginning of the second line
+			offset = doc.getLineOffset(1);
+			event = createDocEvent(offset, SINGLE_SLASH);
+			assertFalse(testEvaluator.evaluate(event));
+
+			// Place another backslash after the previous one
+			offset++;
+			event = createDocEvent(offset, SINGLE_SLASH);
+			// Now the evaluation function should trigger
+			assertTrue(testEvaluator.evaluate(event));
+
+			// Mock a document event with a single backslash placed at the beginning of the first line
+			offset = doc.getLineOffset(0);
+			event = createDocEvent(offset, SINGLE_SLASH);
+			assertFalse(testEvaluator.evaluate(event));
+
+			// Place another backslash after the previous one
+			offset++;
+			event = createDocEvent(offset, SINGLE_SLASH);
 			// Now the evaluation function should trigger
 			assertTrue(testEvaluator.evaluate(event));
 		} catch (BadLocationException e) {
 			// Should never get here
-			// fail included as sanity check
-			fail("BadLocationException thrown in consecutiveLinesUpCommentedOut Test");
+			fail("Should never see this error in: " + this.getClass().getSimpleName() + "::" + this.getClass().getName());
 			e.printStackTrace();
 		}
 	}
+
+	/**
+	 * Verifies that the evaluation function will not trigger when the user types "//" anywhere but the start
+	 * of a line
+	 */
+	@Test
+	public void commentNotAtStartOfLineDown() {
```

Comment:
I think this test might need to be re-worked. We should consider the case where some text is indented on a line. In this case, a user could add two slashes that are after the starting offset of the line but still before any text on the line.

Reply:
Nevermind, I misunderstood the document contents initially